src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -303,6 +303,43 @@ export default class DataTable extends Vue {
     this.$emit(DATA_TABLE_EVENTS.SELECTED_ROWS_CHANGE, selectedRowsData);
   }
 
+  _checkIndeterminate(rowData: DataTableRow) {
+    let parent = rowData.parent;
+
+    while (parent) {
+      const parentCheck = document.querySelector(
+        `#checkbox-select-${this._rowId(parent.rowNumber)}`
+      ) as HTMLInputElement;
+      const children = parent.nestedContent.table.data;
+      const childrenSelected = [];
+
+      children?.forEach(row => {
+        const childCheck = document.querySelector(`#checkbox-select-${this._rowId(row.rowNumber)}`) as HTMLInputElement;
+
+        if (childCheck.checked) {
+          childrenSelected.push(row);
+        }
+      });
+
+      if (childrenSelected.length === 0) {
+        parentCheck.checked = false;
+        parentCheck.classList.remove('-indeterminate');
+        this.deselectRow(parent);
+      } else if (children.length === childrenSelected.length) {
```

Comment:
Checking parent row ccheckbox is optional and project specific

Reply:
I would suggest only setting indeterminate state when SOME of the children are selected. I would not change state `checked`